src/Persistence.Linnapps/Repositories/ProductRangeRepository.cs

```diff
@@ -0,0 +1,56 @@
+﻿namespace Linn.Products.Persistence.Linnapps.Repositories
+{
+    using System;
+    using System.Linq;
+    using System.Linq.Expressions;
+
+    using Linn.Common.Persistence;
+    using Linn.Products.Domain.Linnapps.Products;
+
+    using Microsoft.EntityFrameworkCore;
+
+    public class ProductRangeRepository : IRepository<ProductRange, int>
+    {
+        private readonly ServiceDbContext serviceDbContext;
+
+        public ProductRangeRepository(ServiceDbContext serviceDbContext)
+        {
+            this.serviceDbContext = serviceDbContext;
+        }
+
+        public ProductRange FindById(int key)
+        {
+            return this.serviceDbContext.ProductRanges
+                .Where(b => b.Id == key)
+                .ToList().First();
+        }
+
+        public IQueryable<ProductRange> FindAll()
+        {
+            return this.serviceDbContext.ProductRanges;
+        }
+
+        public void Add(ProductRange entity)
+        {
+            var nextId = this.serviceDbContext.ProductRanges.Max(t => t.Id) + 1;
+            entity.Id = nextId;
```

Comment:
I've not done this either, should this be required for all adds?

Reply:
As above. It's a hack to get the id. It shouldn't be done this way. 
It's only a problem for numeric Id fields. We've got away with it for a few of the forms because the id is a string